### ISSUE TITLE
[cross_entropy]: does not tolerate element [#0 = 4] outside [0, 1]

### ISSUE THREAD

**Issue by grigvardanyan** [microsoft/LightGBM#3986]
## Description
I want to train lgbm classifier with 4 classes. 
Classes: labels are from 1-4.
Data: sparse matrix.
Model params: param = {'num_leaves':31, 'objective':'cross_entropy','num_threads':30}
                         param['metric'] = ['auc','cross_entropy']
Error
![error](https://user-images.githubusercontent.com/26326278/107976331-cc84c280-6fd2-11eb-9e3f-23b678ac33b9.png)

## Environment info

LightGBM version :'3.1.1'

lgbm installation

``` 
pip install lightgbm


```

**Comment by StrikerRUS** [comment]
[`cross_entropy` application](https://lightgbm.readthedocs.io/en/latest/Parameters.html#core-parameters) is not the same as classification. It requires that label values should be in `[0; 1]` range.

> label is anything in interval [0, 1]

You can read https://github.com/microsoft/LightGBM/blob/master/examples/python-guide/logistic_regression.py for more info.
